---
title: "ceratodon pixy plots"
author: "Meng"
date: "2023-02-14"
output: html_document
---

```{r setup, include=FALSE}
library(ggplot2)
library(dplyr)
library(tidyr)
library("stringr")
library(ggvenn)
library(ggVennDiagram)
setwd("/Users/yuanmeng/Dropbox/ceratodon/c2023/")
```

## plot pi along chromosmes

### gene wise pi for auto, U , V

```{r}
pixy <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/pi/auto_allSite_fixWind_pi.txt", header=TRUE) 
pixyU <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/pi/U_allSite_fixWind_pi.txt", header=TRUE) 
pixyV <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/pi/V_allSite_fixWind_pi.txt", header=TRUE) 
pixy <- rbind(pixy, pixyU, pixyV)

pixy <- pixy[complete.cases(pixy),]
mean(pixy$no_sites)
pixy <- pixy %>% filter(no_sites > 100) # see how many you lose, wont be an issue with such big windows

pixy <- pixy %>% mutate(chr_position = ((window_pos_1 + window_pos_2)/2)/1000000)
```
mean(pixy$no_sites) 76771.85


### fixed window pi for auto, U, V
```{r}
ggplot(pixy, aes(x = chr_position, y = avg_pi))+geom_line(size = 0.25)+
    facet_grid(.~chromosome,scales="free_x", switch = "x", space = "free_x") + 
    xlab("Position on Chromosome (Mb)")+ylab("pi")+theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside",legend.position ="none")+ ylim(0, 0.02) 
```


### fix wind or gene wise pi for syn and nonsyn sites
```{r}

```


get g and s specific genes
plot pi - expression ratios
run sfs scripts
i need a break...


## get list of specific genes
```{r}
readcnt<-read.table("/Users/yuanmeng/Dropbox/ceratodon/deseq2/readcnts_ceratodon.filt.txt",header=T)
readcnt<-readcnt[ , -c(2:5)]
head(readcnt)
readcnt[-(1:2)] <- readcnt[-(1:2)]/readcnt[,2][row(readcnt[-(1:2)])]
readcnt[,-c(1:2)] <- apply(readcnt[,-c(1:2)],2, function(x){x * 1e6 / sum(x)})
sum(readcnt[,17])

meta_data<-read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/meta_data_full.csv")
#rownames(meta_data) <- colnames(readcnt_matrix)
#meta_data$id <- seq.int(nrow(meta_data)) 
#l_m <- meta_data %>% filter(sex == "m") %>% select(id)
#l_f <- meta_data %>% filter(sex == "f") %>% select(id)
l_m <- l_m +2
l_f <- l_f +2

write.table(readcnt, file = "readcnt_tpm.txt", sep="\t",row.names = FALSE, col.names = TRUE, quote=FALSE)

readcnt <- readcnt %>% mutate(s_mean=rowMeans(readcnt[,c(3:23)])) %>% mutate(g_mean=rowMeans(readcnt[,c(24:59)])) %>% mutate(g_mean_m=rowMeans(readcnt[,t(l_m)])) %>% mutate(g_mean_f=rowMeans(readcnt[,t(l_f)]))

s_specific_auto <- readcnt %>% filter(s_mean>2 & g_mean<0.5) %>% select(1) 
g_specific_auto <- readcnt %>% filter(g_mean>2 & s_mean<0.5) %>% select(1)  

s_specific_m <- readcnt %>% filter(s_mean>2 & g_mean_m<0.5) %>% select(1) 
g_specific_m <- readcnt %>% filter(g_mean_m>2 & s_mean<0.5) %>% select(1)  

s_specific_f <- readcnt %>% filter(s_mean>2 & g_mean_f<0.5) %>% select(1) 
g_specific_f <- readcnt %>% filter(g_mean_f>2 & s_mean<0.5) %>% select(1)  

# for gene on U and V, g sample used for comparison should be only females or males
# readcnt <- readcnt %>% mutate(s_mean=rowMeans(readcnt[,c(3:23)])) %>% mutate(g_mean=rowMeans(readcnt[,c(24:59)]))
# s_specific <- readcnt %>% filter(s_mean>2 & g_mean<0.5) %>% select(1) 
# g_specific <- readcnt %>% filter(g_mean>2 & s_mean<0.5) %>% select(1)  

gff<-read.table("/Users/yuanmeng/Dropbox/ceratodon/deseq2/R40_GG1U_combined.gene.final.gff")
head(gff)
colnames(gff)<-c("chrom","start","end","gene")

s_specific <- s_specific_auto 
colnames(s_specific)<-"gene"
s_specific <- left_join(s_specific,gff,by="gene")
s_specific <- s_specific[complete.cases(s_specific),]
s_specific <- s_specific %>% filter(chrom != "U" & chrom !="V")
write.table(s_specific[,c(2:4)],file="s_specific_auto.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)

s_specific <- s_specific_f
colnames(s_specific)<-"gene"
s_specific <- left_join(s_specific,gff,by="gene")
s_specific <- s_specific[complete.cases(s_specific),]
s_specific <- s_specific %>% filter(chrom == "U")
write.table(s_specific[,c(2:4)],file="s_specific_U.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)

s_specific <- s_specific_m
colnames(s_specific)<-"gene"
s_specific <- left_join(s_specific,gff,by="gene")
s_specific <- s_specific[complete.cases(s_specific),]
s_specific <- s_specific %>% filter(chrom == "V")
write.table(s_specific[,c(2:4)],file="s_specific_V.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)

g_specific <- g_specific_auto 
colnames(g_specific)<-"gene"
g_specific <- left_join(g_specific,gff,by="gene")
g_specific <- g_specific[complete.cases(g_specific),]
g_specific <- g_specific %>% filter(chrom != "U" & chrom !="V")
write.table(g_specific[,c(2:4)],file="g_specific_auto.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)

g_specific <- g_specific_f
colnames(g_specific)<-"gene"
g_specific <- left_join(g_specific,gff,by="gene")
g_specific <- g_specific[complete.cases(g_specific),]
g_specific <- g_specific %>% filter(chrom == "U")
write.table(g_specific[,c(2:4)],file="g_specific_U.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)

g_specific <- g_specific_m
colnames(g_specific)<-"gene"
g_specific <- left_join(g_specific,gff,by="gene")
g_specific <- g_specific[complete.cases(g_specific),]
g_specific <- g_specific %>% filter(chrom == "V")
write.table(g_specific[,c(2:4)],file="g_specific_V.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)

```
s specific auto 578
s specific U 84
s specific V 41
g specific auto 580
g specific U 99
g specific V 117
no bias in these ratios, need to redo the plot and fisher's exact test
 
 

### get combined g and s specific gene
compare the list genes specific genes you get from using tpm vs normalized read counts
for DFE we used tom cutoffs
for ratio expression plots we used normalized read count
the result should be the same?
```{r}
g_specific <- read.table("g_specific.bed")
s_specific <- read.table("s_specific.bed")
colnames(g_specific)<-c("chrom","start","end")
colnames(s_specific)<-c("chrom","start","end")
g_specific <- inner_join(g_specific,gff,by=c("chrom","start","end"))
s_specific <- inner_join(s_specific,gff,by=c("chrom","start","end"))

write.table(s_specific, file="s_specific.txt",sep="\t",row.names = FALSE, quote=FALSE)
write.table(g_specific, file="g_specific.txt",sep="\t",row.names = FALSE, quote=FALSE)
```
g specific 2129
s specific 2014

### getting cnts and s_g_ratios
need to get g_mean_m for V genes and g_mean_f for U genes
```{r}
cnt <-read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/normalized_counts.csv")
# s 2:22, g 23:58, g_m 
# normalized_counts<-mutate(normalized_counts, s_mean = rowMeans(normalized_counts[,2:22]))
# normalized_counts<-mutate(normalized_counts, g_mean = rowMeans(normalized_counts[,23:58]))
# %>% mutate(g_mean_m=rowMeans(readcnt[,t(l_m)])) %>% mutate(g_mean_f=rowMeans(readcnt[,t(l_f)]))
l_f <- l_f - 1
l_m <- l_m - 1
cnt <- cnt %>% mutate(s_mean = rowMeans(cnt[,2:22])) %>% 
    mutate(g_mean = rowMeans(cnt[,23:58])) %>% 
    mutate(g_mean_m = rowMeans(cnt[,t(l_m)])) %>% 
    mutate(g_mean_f = rowMeans(cnt[,t(l_f)]))

DE_tissue_pollen$gene <- 
# s_g_ratio, pollenTube_leaf_ratio
# normalized_counts<-mutate(normalized_counts, s_g_ratio = s_mean/g_mean)
cnt <- cnt %>% mutate(s_g_ratio = s_mean/g_mean) %>% 
    mutate(s_gm_ratio = s_mean/g_mean_m) %>% 
    mutate(s_gf_ratio = s_mean/g_mean_f)

colnames(cnt)[1] <- "gene"
cnt <- inner_join(cnt, gff, by="gene")

cnt_auto <- mutate(cnt, Chrm = str_sub(cnt$chrom, 1, 1)) %>% filter(Chrm == "C") %>% select(gene,chrom,start,end, s_g_ratio)

cnt_U <- cnt %>% filter(chrom == "U") %>% select(gene,chrom,start,end, s_gf_ratio)
cnt_V <- cnt %>% filter(chrom == "V") %>% select(gene,chrom,start,end, s_gm_ratio)
colnames(cnt_V)[5] <- "s_g_ratio"

cnt <- rbind(cnt_auto,cnt_U,cnt_V)
cnt <- cnt[complete.cases(cnt),]
write.table(cnt, file = "cnt_ratios_final.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)


cnt<- cnt %>% select(c(1,59:65))
head(cnt)
write.table(cnt[,c(1,59:65)], file = "normalized_counts_ratios.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)
```
cnt 30491

### merge the s_g_ratio dataset with the tpm-idetified specific genes
```{r}
head(g_specific)
g_specific <- mutate(g_specific, category = "g_specific_tpm")
s_specific <- mutate(s_specific, category = "s_specific_tpm")

head(g_specific_ratio)
g_specific_ratio <- cnt %>% filter(s_g_ratio == 0)
s_specific_ratio <- cnt %>% filter(is.infinite(s_g_ratio))
    
g_specific_full <- full_join(g_specific, g_specific_ratio, by = c("chrom", "start", "end", "gene"))
s_specific_full <- full_join(s_specific, s_specific_ratio, by = c("chrom", "start", "end", "gene"))

#get a venn diagram here
v1<-g_specific$gene
v2<-s_specific$gene
v3<-g_specific_ratio$gene
v4<-s_specific_ratio$gene
venn.diagram(
  x = list(v1, v2, v3, v4),
  category.names = c("g_specific_tpm" , "s_specific_tpm" , "g_specific_ratio", "s_specific_ratio"),
  filename = NULL,
  output=TRUE,
  cat.cex = 1,
  cat.fontface = "bold")

# intersect these genes with significant DF genes
g_bias_auto <- read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/gametophyte_biased_gene.csv") %>% filter(chrom != "U" & chrom != "V")
g_bias_U <- read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/gametophyte_biased_gene_Nov14female.csv") %>% filter(chrom == "U")
g_bias_V <- read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/gametophyte_biased_gene_Nov14male.csv") %>% filter(chrom == "V")
s_bias_auto <- read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/sporophyte_biased_gene.csv") %>% filter(chrom != "U" & chrom != "V")
s_bias_U <- read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/sporophyte_biased_gene_Nov14female.csv") %>% filter(chrom == "U")
s_bias_V <- read.csv("/Users/yuanmeng/Dropbox/ceratodon/deseq2/sporophyte_biased_gene_Nov14male.csv") %>% filter(chrom == "V")

write.csv(g_bias_auto, file = "g_bias_auto.csv", row.names = FALSE, quote = FALSE)
write.csv(g_bias_U, file = "g_bias_U.csv", row.names = FALSE, quote = FALSE)
write.csv(g_bias_V, file = "g_bias_V.csv", row.names = FALSE, quote = FALSE)
write.csv(s_bias_auto, file = "s_bias_auto.csv", row.names = FALSE, quote = FALSE)
write.csv(s_bias_U, file = "s_bias_U.csv", row.names = FALSE, quote = FALSE)
write.csv(s_bias_V, file = "s_bias_V.csv", row.names = FALSE, quote = FALSE)

g_bias <- rbind(g_bias_auto,g_bias_U,g_bias_V)
s_bias <- rbind(s_bias_auto,s_bias_U,s_bias_V)
write.csv(g_bias, file = "g_bias.csv", row.names = FALSE, quote = FALSE)
write.csv(s_bias, file = "s_bias.csv", row.names = FALSE, quote = FALSE)




g_ratio <- intersect(g_bias$gene, g_specific_ratio$gene) # 247
g_tpm <- intersect(g_bias$gene, g_specific$gene) # 776
g_ratio_tpm <- intersect(g_ratio,g_tpm) # 123
# report to stephen
s_ratio <- intersect(s_bias$gene, s_specific_ratio$gene) # 65
s_tpm <- intersect(s_bias$gene, s_specific$gene) # 656
s_ratio_tpm <- intersect(s_ratio,s_tpm) # 34
# does this mean a larger agreement??
# slightly larger agreement
venn.diagram(
  x = list(g_tpm, s_tpm, g_ratio, s_ratio),
  category.names = c("g_specific_tpm" , "s_specific_tpm" , "g_specific_ratio", "s_specific_ratio"),
  filename = NULL,
  output=TRUE,
  cat.cex = 1,
  cat.fontface = "bold")

v1<-g_specific$gene  # tpm
v2<-s_specific$gene
v3<-g_specific_ratio$gene
v4<-s_specific_ratio$gene

x <- list(g_tpm=g_tpm, g_ratio=g_ratio, s_ratio=s_ratio, s_tpm=s_tpm)
x <- list(g_tpm=v1, g_ratio=v3, s_ratio=v4, s_tpm=v2)
ggvenn(x,fill_color = c("white","white","white","white"),
  stroke_size = 0.5, set_name_size = 5,show_percentage = FALSE) 

write.table(g_ratio, file = "g_specific_ratio_deseq2.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)
write.table(s_ratio, file = "s_specific_ratio_deseq2.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)
write.table(g_tpm, file = "g_specific_tpm_deseq2.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)
write.table(s_tpm, file = "s_specific_tpm_deseq2.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)

# use g_ratio and s_ratio moving forward
# redo deseq2, keep original unfiltered DE results on save
```
g specific 796
s specific 703
g_specific_ratio 1262
s_specific_ratio 602
g_specific_full 1929
s_specific_full 1255
g_bias 7619
s_bias 4921

should filter on readcnt (min readcnt for all samples>20?) then redo nomalized read cnt or just refine the set of normalize read cnt

get list of genes statistically DE between tissues and baseMean>1 for identifying specific genes
or also read count across samples >10? (basially baseMean>1)??





## plot pi and expression ratios, add g and s specific genes
### getting ratio, gene sets
```{r}
cnt <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/cnt_ratios_final.txt",header = TRUE)
g_specific <- as.data.frame(g_ratio)
s_specific <- as.data.frame(s_ratio)
# use slightly bigger set to repeat the anlyses and plots
colnames(g_specific)[1] <- "gene"
colnames(s_specific)[1] <- "gene"
# get clean sets
g_specific <- g_specific %>% mutate(category = "g_specific")
s_specific <- s_specific %>% mutate(category = "s_specific")
specific <- rbind(g_specific,s_specific)

cnt <- left_join(cnt,specific, by="gene")

cnt_clean <- cnt %>% filter(s_g_ratio>0 & is.finite(s_g_ratio)) # 30491-> 28627
# redundant step, already taken care of
# cnt_clean <- cnt_clean %>% filter(category != "g_specific" & category != "s_specific")

g_specific <- g_specific %>% mutate(s_g_ratio = 0)
s_specific <- s_specific %>% mutate(s_g_ratio = 10000000)
specific <- inner_join(specific, gff, by="gene")
write.table(specific, file = "specific_genes_Feb28.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE) # p005
write.table(specific, file = "specific_genes_Mar21.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE) # p01
```
get expressed genes both stages, and g s specific genes subsets
for linear model only use the genes expression in both stages, not including specific genes
redo this for rumex dataset for the manuscript
```{r}
# get a subset for binning quantiles
ratio_pixy_clean <- ratio_pixy %>% filter(s_g_ratio>0)  %>% filter(is.finite(s_g_ratio)) 

# ratio_pixy[which(!is.finite(ratio_pixy$pollen_leaf_ratio)),] <- 10000000
#ratio_pixy <- do.call(data.frame, lapply(ratio_pixy,function(x) replace(x, is.infinite(x), 10000000)))
```



### getting stats (pi) 
separate code for tajD?

filter on number of sites per gene
```{r}
pixyA <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/pi/auto_syn_geneWise_pi.txt", header=TRUE) 
pixyU <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/pi/U_syn_geneWise_pi.txt", header=TRUE) 
pixyV <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/pi/V_syn_geneWise_pi.txt", header=TRUE) 
pixy <- rbind(pixyA, pixyU, pixyV) #25100
pixy <- pixy[complete.cases(pixy),]
head(pixy)

# see how many are left
pixyA <- pixyA %>% filter(no_sites > 50) 
pixyU <- pixyU %>% filter(no_sites > 50) 
pixyV <- pixyV %>% filter(no_sites > 50) 

colnames(pixy)[2:4] <- c("chrom", "start", "end")
ratio_pixy<- inner_join(cnt_clean, pixy, by=c("chrom", "start", "end"))  # see how many left
write.csv(ratio_pixy, file = "ratio_pixy_Feb28.csv", row.names = FALSE, quote = FALSE)
ratio_pixy <- read.csv("ratio_pixy_Feb28.csv")

# can relax on no_sites criteria to have more genes, come back later
ratio_pixy_specific<- inner_join(specific, pixy, by=c("chrom", "start", "end"))  # see how many 
nrow(filter(ratio_pixy_specific,category=="g_specific"))

ratio_pixy_full <- rbind(ratio_pixy,ratio_pixy_specific)
write.csv(ratio_pixy_full, file = "ratio_pixy_full_Feb28.csv", row.names = FALSE, quote = FALSE)
write.csv(ratio_pixy_full, file = "ratio_pixy_full_Mar21.csv", row.names = FALSE, quote = FALSE)
```
pi A: 27598 - 21233 (minimum 50 sites per gene)
Pi U: 3450 - 2152
pi V: 3411 - 1715
pixy 25100
cnt_clean 28627 (not secific) 
ratio_pixy 22305
speciifc 312 -> 329 (only 17 more genes)
ratio_pixy_specific 158 (g118, s40) -> 170 (g128, s42)

### linear regression for genes expressed in both life stages
can wait, after we know the criteria to subset the genes

```{r}
ratio_pixy$log_ratio<-log10(ratio_pixy$s_g_ratio)
ratio_pixy$abs_ratio<-abs(log10(ratio_pixy$s_g_ratio))
#ratio_pixy <- do.call(data.frame, lapply(ratio_pixy,function(x) replace(x, is.infinite(x), -10000000)))
#ratio_pixy3<-ratio_pixy%>% filter(abs_ratio>-10)

ggplot(ratio_pixy, aes(x=log10(s_g_ratio), y=avg_pi)) + geom_point(colour="#1F78B4", alpha=.3) + geom_smooth(color="black") + theme_classic() + xlim(-2, 2) + labs(y="", x="") + ylim(0,0.06) + theme(text = element_text(size = 30))

ggplot(ratio_pixy, aes(x=abs(log10(s_g_ratio)), y=avg_pi)) + geom_point(colour="#1F78B4", alpha=.3) + geom_smooth(color="black") + theme_classic() + labs(y="", x="") + ylim(0,0.06) + theme(text = element_text(size = 40))
```


```{r}
# slope for linear regression
# significant test
ratio.lm = lm(avg_pi ~ log_ratio, data=ratio_pixy)
summary(ratio.lm) 

ratio.lm = lm(avg_pi ~ abs_ratio, data=ratio_pixy)
summary(ratio.lm) 

ratio_pixy2$log_ratio<-log10(ratio_pixy2$pollen_leaf_ratio)
ratio_pixy2$abs_ratio<-abs(log10(ratio_pixy2$pollen_leaf_ratio))
ratio_pixy2<-ratio_pixy2%>% filter(log_ratio>-3)%>% filter(log_ratio<3)
```

anova test between the categories 
(g128, s42)
```{r}
ratio_pixy <- ratio_pixy %>% mutate(category = "expressed")
ratio_pixy_full <- rbind(ratio_pixy,ratio_pixy_specific)
ratio_pixy_full
aov.model <- aov(ratio_pixy_full$avg_pi ~ ratio_pixy_full$category)
summary(aov.model)

lm.model <- lm(ratio_pixy_full$avg_pi ~ ratio_pixy_full$category)
summary(lm.model)
```

Df Sum Sq   Mean Sq F value Pr(>F)
ratio_pixy_full$category     2  0.000 0.0001322   0.753  0.471
Residuals                22472  3.946 0.0001756   

Call:
lm(formula = ratio_pixy_full$avg_pi ~ ratio_pixy_full$category)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.012884 -0.009653 -0.003912  0.005456  0.140102 

Coefficients:
                                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)                         1.233e-02  8.873e-05 138.999   <2e-16 ***
ratio_pixy_full$categoryg_specific  5.501e-04  1.175e-03   0.468    0.640    
ratio_pixy_full$categorys_specific -2.318e-03  2.047e-03  -1.133    0.257    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.01325 on 22472 degrees of freedom
Multiple R-squared:  6.7e-05,	Adjusted R-squared:  -2.199e-05 
F-statistic: 0.7529 on 2 and 22472 DF,  p-value: 0.471



### plot with quantiles
can wait till we figure on the gene set criteria
```{r}
bin<-binning(ratio_pixy$s_g_ratio, bins=8, method="quantile",
                     labels=NULL, ordered=TRUE, weights=NULL)
plot(bin)

# set up cut-off values 
breaks <- c(-1, 0.001, 0.234,0.443,0.654,0.881,1.17,1.62,3.07,1000000,100000000000000)
# specify interval/bin labels
tags <- c("0","(0-0.234]","(0.234-0.443]", "(0.443-0.654]", "(0.654,0.881]","(0.881-1.17]", "(1.17-1.62]", "(1.62-3.07]", "[3.07-inf)","inf")

# bucketing values into bins
group_tags <- cut(ratio_pixy_full$s_g_ratio, 
                  breaks=breaks, 
                  include.lowest=TRUE, 
                  right=FALSE, 
                  labels=tags)
# inspect bins
summary(group_tags)

ratio_groups <- factor(group_tags, 
                           levels = tags,
                           ordered = TRUE)
# the real graph 0,0.00102,0.117,0.224,0.4,0.802,1.82,5.21,27.8
v <- ratio_pixy_full %>% select(s_g_ratio,avg_pi) #pick the variable 

vgroup <- as_tibble(v) %>% 
  mutate(tag = case_when(
    s_g_ratio == 0 ~ tags[1], 
    s_g_ratio > 0 & s_g_ratio <= 0.234 ~ tags[2],
    s_g_ratio > 0.234 & s_g_ratio <= 0.443 ~ tags[3],
    s_g_ratio > 0.443 & s_g_ratio <= 0.654 ~ tags[4],
    s_g_ratio > 0.654 & s_g_ratio <= 0.881 ~ tags[5],
    s_g_ratio > 0.881 & s_g_ratio <= 1.17 ~ tags[6],
    s_g_ratio > 1.17 & s_g_ratio <= 1.62 ~ tags[7],
    s_g_ratio > 1.62 & s_g_ratio <= 3.07 ~ tags[8],
    s_g_ratio > 3.07 & s_g_ratio < 10000000 ~ tags[9],
    s_g_ratio == 10000000 ~ tags[10]
    ))
summary(vgroup)
vgroup$tag <- factor(vgroup$tag,
                       levels = tags,
                       ordered = FALSE)

ggplot(data = vgroup, mapping = aes(x=tag,y=avg_pi)) + 
  geom_jitter(color="#1F78B4",alpha=0.2) +
  geom_boxplot(fill="#1F78B4",color="black",alpha=0.3) + 
  labs(x='s_g_ratio') +
  guides(color=FALSE) + ylim(0,0.06)+
  theme_classic() 
```



## flip ratio to g_s_ratio (not s_g_ratio)
use old approach where there's no special cutoff for specific genes
```{r}
cnt <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/normalized_counts_ratios.txt",header=T)
# cnt2 <- read.table("/Users/yuanmeng/Dropbox/ceratodon/c2023/cnt_ratios_final.txt",header=T)
# or just re read the gff file
gff<-read.table("/Users/yuanmeng/Dropbox/ceratodon/deseq2/R40_GG1U_combined.gene.final.gff")
colnames(gff)<-c("chrom","start","end","geneID")

cnt <- cnt %>% mutate(g_s_ratio = g_mean/s_mean) %>% 
    mutate(gm_s_ratio = g_mean_m/s_mean) %>% 
    mutate(gf_s_ratio = g_mean_f/s_mean)

cnt <- inner_join(cnt, gff, by="geneID")

cnt_auto <- mutate(cnt, Chrm = str_sub(cnt$chrom, 1, 1)) %>% filter(Chrm == "C") %>% select(geneID,chrom,start,end, g_s_ratio)
cnt_U <- cnt %>% filter(chrom == "U") %>% select(geneID,chrom,start,end, gf_s_ratio)
cnt_V <- cnt %>% filter(chrom == "V") %>% select(geneID,chrom,start,end, gm_s_ratio)
colnames(cnt_V)[5] <- "g_s_ratio"
colnames(cnt_U)[5] <- "g_s_ratio"
cnt_final <- rbind(cnt_auto,cnt_U,cnt_V)

cnt_final <- cnt[complete.cases(cnt_final),]
write.table(cnt_final, file = "/Users/yuanmeng/Dropbox/ceratodon/c2023/cnt_ratios_final_Apr16.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)
write.table(cnt, file = "normalized_counts_ratios_Apr16.txt", sep="\t", row.names = FALSE, col.names = TRUE, quote=FALSE)

```



```{r}
cnt_clean <- cnt_final %>% filter(g_s_ratio>0 & is.finite(g_s_ratio)) # 30491-> 28627 (30917-26692)
ratio_pixy <- inner_join(cnt_final, pixy, by=c("chrom", "start", "end")) %>% select(geneID,chrom,start,end, g_s_ratio, avg_pi) # 22706

ratio_pixy_clean <-ratio_pixy %>% filter(g_s_ratio>0 & is.finite(g_s_ratio))

write.csv(ratio_pixy, file = "/Users/yuanmeng/Dropbox/ceratodon/c2023/ratio_pixy_Apr16.csv", row.names = FALSE, quote = FALSE)

ratio_pixy <- ratio_pixy[complete.cases(ratio_pixy),]
# what to do specific genes here?
# we never decided how to present this graph
# maybe do both
bin<-binning(ratio_pixy_clean$g_s_ratio, bins=8, method="quantile",
             labels=NULL, ordered=TRUE, weights=NULL)
plot(bin)

# set up cut-off values 
breaks <- c(-1, 0.001, 0.234,0.443,0.654,0.881,1.17,1.62,3.07,1000000,100000000000000)
# specify interval/bin labels
tags <- c("0","(0-0.234]","(0.234-0.443]", "(0.443-0.654]", "(0.654,0.881]","(0.881-1.17]", "(1.17-1.62]", "(1.62-3.07]", "[3.07-inf)","inf")

# bucketing values into bins
#group_tags <- cut(ratio_pixy$g_s_ratio, 
                  # breaks=breaks, 
                  # include.lowest=TRUE, 
                  # right=FALSE, 
                  # labels=tags)
# inspect bins
#summary(group_tags)
#ratio_groups <- factor(group_tags, 
                       # levels = tags,
                       # ordered = TRUE)
# the real graph 0,0.00102,0.117,0.224,0.4,0.802,1.82,5.21,27.8
v <- ratio_pixy %>% select(g_s_ratio,avg_pi) #pick the variable 
# [5.41e-06,0.306]
# (0.582.0.797]
# (1.05,1.38]
# (2.01,3.88]

vgroup <- as_tibble(v) %>% 
    mutate(tag = case_when(
        g_s_ratio == 0 ~ tags[1], 
        g_s_ratio > 0 & g_s_ratio <= 0.306 ~ tags[2],
        g_s_ratio > 0.306 & g_s_ratio <= 0.582 ~ tags[3],
        g_s_ratio > 0.582 & g_s_ratio <= 0.797 ~ tags[4],
        g_s_ratio > 0.797 & g_s_ratio <= 1.05 ~ tags[5],
        g_s_ratio > 1.05 & g_s_ratio <= 1.38 ~ tags[6],
        g_s_ratio > 1.38 & g_s_ratio <= 2.01 ~ tags[7],
        g_s_ratio > 2.01 & g_s_ratio <= 3.88 ~ tags[8],
        g_s_ratio > 3.88 & g_s_ratio <= 42500 ~ tags[9],
        is.infinite(g_s_ratio) ~ tags[10]
    ))

summary(vgroup)
vgroup$tag <- factor(vgroup$tag,
                     levels = tags,
                     ordered = FALSE)

ggplot(data = vgroup, mapping = aes(x=tag,y=avg_pi)) + 
    geom_jitter(color="#1F78B4",alpha=0.2) +
    geom_boxplot(fill="#1F78B4",color="black",alpha=0.3) + 
    labs(x='g_s_ratio') +
    guides(color=FALSE) + ylim(0,0.06)+
    theme_classic() + theme(text = element_text(size = 36))

```

save the ratio genes as new specific genes
```{r}
ss <- cnt_final %>% filter(g_s_ratio == 0) %>% select(chrom, start, end) # 532
gs <- cnt_final %>% filter(is.infinite(g_s_ratio)) %>% select(chrom, start, end)  #1217
# write .bed
write.table(ss,file="ss_ratio.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)
write.table(gs,file="gs_ratio.bed",sep="\t",row.names = FALSE, col.names = FALSE, quote=FALSE)
```

worth trying tpm???
```{r}

```

