---
title: "balancing selection in rumex"
author: "Meng"
date: "2023-11-27"
output: html_document
---

```{r setup, include=FALSE}
library(dplyr)
library(ggplot2)
library(patchwork)
library(tidyr)
data="/Users/yuanmeng/Library/CloudStorage/OneDrive-UniversityofToronto/Manuscripts/rumex_haploid_sel/balsel/"
```

## plot finished scans

read data
```{r}
scan1 <- read.table(paste0(data,"rumex_scan_DAF_Jul6_LG1.txt"), header = T)
scan2 <- read.table(paste0(data,"rumex_scan_DAF_Jul6_LG2.txt"), header = T)
scan3 <- read.table(paste0(data,"rumex_scan_DAF_Jul6_LG3.txt"), header = T)
scan4 <- read.table(paste0(data,"rumex_scan_DAF_Jul6_LG4.txt"), header = T)
scan5 <- read.table(paste0(data,"rumex_scan_DAF_Jul6_LG5.txt"), header = T)
scan1$chrom <- "LG1"
scan2$chrom <- "LG2"
scan3$chrom <- "LG3"
scan4$chrom <- "LG4"
scan5$chrom <- "LG5"

scan <- rbind(scan1, scan2, scan3, scan4, scan5)
scan<- scan %>% mutate(chrom = factor(chrom))
levels(scan$chrom)
levels(scan$chrom)<- c("A1", "A4", "A2", "XY", "A3")
#scan <- scan[order(scan$chrom),]
scan <- scan %>% mutate(chrom = factor(chrom, levels = c("A1","A2","A3","A4","XY")))
```

## plot inversion on A2, A3

```{r}
scan3$chrom <- "A2"
scan5$chrom <- "A3"
scan2$chrom <- "A4"
p2 <- ggplot(scan5, aes(x = physPos/1000000, y = CLR))+geom_point(aes(colour=log10(s_hat)),size = 0.3) + ggtitle("A3") +
    xlab("Physical position on Chromosome (Mb)")+ylab("Composite Likelihood Ratio")+theme_bw() +scale_color_gradient2(name = expression('log'[10]~hat(italic(a))), limits = c(-3, 9), midpoint = 0, high="red", mid = 'white', low="blue")+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside")+ theme(text = element_text(size = 10), panel.spacing = unit(0.5, "lines")) + geom_vline(xintercept = c(11.5,31.8))

p1 <- ggplot(scan3, aes(x = physPos/1000000, y = CLR))+geom_point(aes(colour=log10(s_hat)),size = 0.3) + ggtitle("A2") +
    xlab("Physical position on Chromosome (Mb)")+ylab("Composite Likelihood Ratio")+theme_bw() +scale_color_gradient2(name = expression('log'[10]~hat(italic(a))), limits = c(-3, 9), midpoint = 0, high="red", mid = 'white', low="blue")+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside")+ theme(text = element_text(size = 10), panel.spacing = unit(0.5, "lines")) + geom_vline(xintercept = c(5.6,69))

p3 <- ggplot(scan2, aes(x = physPos/1000000, y = CLR))+geom_point(aes(colour=log10(s_hat)),size = 0.3) + ggtitle("A4") +
    xlab("Physical position on Chromosome (Mb)")+ylab("Composite Likelihood Ratio")+theme_bw() +scale_color_gradient2(name = expression('log'[10]~hat(italic(a))), limits = c(-3, 9), midpoint = 0, high="red", mid = 'white', low="blue")+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside")+ theme(text = element_text(size = 10), panel.spacing = unit(0.5, "lines")) + geom_vline(xintercept = c(95,115))


p1 + p2 + p3 + plot_layout(guides = "collect") & theme(legend.position = "right")
```


plot the scans
```{r}
lines_data_facet_A2 <- data.frame(xintercept = c(5.6,69), facet = "A2")
lines_data_facet_A3 <- data.frame(xintercept = c(0,34), facet = "A3")
lines_data_facet_A4 <- data.frame(xintercept = c(95,115), facet = "A4")


p7 <- ggplot(scan, aes(x = physPos/1000000, y = genPos))+
    facet_grid(.~chrom,scales="free_x", switch = "x", space = "free_x") + geom_point(size=0.3, alpha=0.5)+ ggtitle("a") +
    xlab("")+ylab("Genetic position (cM)")+theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside",legend.position ="none")+ theme(text = element_text(size = 10), panel.spacing = unit(0.5, "lines"))

p8 <- ggplot(scan, aes(x = physPos/1000000, y = CLR))+geom_point(aes(colour=log10(s_hat)),size = 0.3)+
    facet_grid(.~chrom,scales="free_x", switch = "x", space = "free_x") + ggtitle("b")+
    xlab("Physical position on Chromosome (Mb)")+ylab("Composite Likelihood Ratio")+theme_bw() +scale_color_gradient2(name = expression('log'[10]~hat(italic(a))), limits = c(-3, 9), midpoint = 0, high="red", mid = 'white', low="blue")+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside")+ theme(text = element_text(size = 10), panel.spacing = unit(0.5, "lines")) + geom_vline(data = lines_data_facet_A2, aes(xintercept = xintercept)) +
  geom_vline(data = lines_data_facet_A3, aes(xintercept = xintercept))+
  geom_vline(data = lines_data_facet_A4, aes(xintercept = xintercept))

#+ geom_hline(yintercept = 9.500813, linetype="dashed",color = "black", size =0.3)


p7 / p8 + plot_layout(guides = "collect") & theme(legend.position = "right")
```


## get 95% of CLR
```{r}
quantile(scan$CLR, probs = 0.95)
summary(scan$CLR)

scan0 <- filter(scan, CLR>0)
nrow(scan0)

quantile(scan0$CLR, probs = 0.95)
summary(scan0$CLR)

scan95 <- filter(scan0, CLR>9.500813)
nrow(scan95)
write.table(scan95, file=paste0(data,"balsel_scan95.txt"), row.names = FALSE, quote = FALSE)
scan95 <- read.table(paste0(data,"balsel_scan95.txt"), header = T)

table(scan95$chrom)[1:5]
```
95% 
5.367076 

     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
   0.0000    0.0000    0.0000    1.5090    0.8816 2558.8470 
4945561 sites tested
240029 sites in 95% quantile
1621040 with non-zero CLR
79431 sites in the 95% quantile of non-zero CLR
31882 in genic regions, 449 genes
155 significant pollen biased genes

     95% 
9.500813 
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
   0.0000    0.8816    1.9159    4.6037    4.2158 2558.8470 
   
## genes in inversions
```{r}
scan95 <- scan95 %>% mutate(physPos_mb = physPos/1000000)
inversion_A2 <- scan95 %>% filter(chrom == "A2" & physPos_mb > 5.6 & physPos_mb < 69) # 10055
# 
inversion_A3 <- scan95 %>% filter(chrom == "A3" & physPos_mb < 34) # 11132
inversion_A4 <- scan95 %>% filter(chrom == "A4" & physPos_mb > 95 & physPos_mb < 115) # 121
inversion <- rbind(inversion_A2, inversion_A3, inversion_A4)

inversion <- inversion %>% inner_join(gff, by = "chrom") %>%
        filter(physPos >= start & physPos <= end) # 9750

result_CLR <- inversion %>%
    group_by(gene) %>%
    summarise(mean_CLR = mean(CLR), max_CLR = max(CLR), no_sites = length(CLR)) # 138

result_CLR <- left_join(result_CLR, gff, by = "gene")
table(result_CLR$chrom)[1:5]
result_tissue <- read.table(paste0(directory,"result_tissue_full.txt"), header=T)
result_tissue <- result_tissue %>% filter(l_mean > 5 | p_mean > 5)
result_CLR <- left_join(result_CLR, result_tissue, by = "gene")

## 
write.table(result_CLR, file=paste0(data,"inversion_genes.txt"), row.names = FALSE, quote = FALSE)
write.table(inversion, file=paste0(data,"inversion_sites_gff.txt"), row.names = FALSE, quote = FALSE)
inversion2 <- rbind(inversion_A2, inversion_A3, inversion_A4)
write.table(inversion2, file=paste0(data,"inversion_sites_all.txt"), row.names = FALSE, quote = FALSE)
```
A1 A4 A2 XY A3 
 0  1 63  0 61 /74
 

## get the list of candidate genes
perform GO enrichment of these candidate genes
how many of these genes are gemetophyte-expressed / gametophyte-biased? what are the functions?
```{r}
directory="/Users/yuanmeng/Library/CloudStorage/OneDrive-UniversityofToronto/Manuscripts/rumex_haploid_sel/deseq2/"
gff <- read.table(paste0(directory,"REF_LA_filtered_annotation.genes.gff"))
colnames(gff) <- c("gene","chrom", "start", "end")

gff<- gff %>% mutate(chrom = factor(chrom))
levels(gff$chrom)[1:5]
levels(gff$chrom)[1:5]<- c("A1", "A4", "A2", "XY", "A3")

# result <- scan95 %>% inner_join(gff, by = "chrom") %>%
#     filter(physPos >= start & physPos <= end) 

window_size <- 7000
num_windows <- floor(nrow(scan95) / window_size) # 11
results <- list()
for (i in 1:12) {
    start <- (i - 1) * window_size + 1
    end <- i * window_size
    window_df <- scan95[start:end, ]
    
    result <- window_df %>% inner_join(gff, by = "chrom") %>%
        filter(physPos >= start & physPos <= end) 
    
    results[[i]] <- result}

result <- do.call(rbind, results)

write.table(result, file=paste0(data,"balsel_sites.txt"), row.names = FALSE, quote = FALSE)
result <- read.table(paste0(data,"balsel_sites.txt"), header = T)

result_CLR <- result %>%
    group_by(gene) %>%
    summarise(mean_CLR = mean(CLR), max_CLR = max(CLR), no_sites = length(CLR))

result_tissue <- read.table(paste0(directory,"result_tissue_full.txt"), header=T)
result_tissue <- result_tissue %>% filter(l_mean > 5 | p_mean > 5)

genes <- left_join(result_CLR, result_tissue, by = "gene")


table(genes$chrom)[1:5]
write.table(genes, file=paste0(data,"balsel_genes_expressed.txt"), sep = "\t",row.names = FALSE, quote = FALSE)
genes <- read.table(paste0(data,"balsel_genes_expressed.txt"), header =T)
genes <- genes %>% filter(l_mean > 5 | p_mean > 5)


pollen_bias <- result_tissue %>% filter(log2FoldChange>0 & padj < 0.1) # 6823
unbias <- DE_tissue %>% filter(log2FoldChange <= 1.02 & log2FoldChange > -1.27) # 4344
unbias <- unbias %>% filter(padj >= 0.1) # 822
leaf_bias <- result_tissue %>% filter(log2FoldChange<0 & padj < 0.1) # 10301


pollen_bias <- genes %>% filter(log2FoldChange>0 & padj < 0.1) # 142
unbiased <- genes %>% filter(padj >= 0.1) # 21
unbiased <- genes %>% filter(log2FoldChange <= 1.02 & log2FoldChange > -1.27 & padj >= 0.1) # 21
leaf_bias <- genes %>% filter(log2FoldChange<0 & padj < 0.1) # 242
pollen_bias$category <- "pollen_bias"
unbiased$category <- "unbiased"
leaf_bias$category <- "leaf_bias"
bias <- rbind(pollen_bias,unbiased,leaf_bias)

write.table(bias, file=paste0(data,"balsel_genes_bias.txt"), row.names = FALSE, quote = FALSE)
```
category, p_biased, unbiased, l_biased
balancing_sel, 142, 21, 242
whole_genome, 6823, 822, 10301


LG1 LG2 LG3 LG4 LG5 
106  32 119  64  84 

A1 A4 A2 XY A3 
43  6 50 27 29 

## check for disease resistance genes
lift the expressed criteria here
```{r}
genes <- read.table(paste0(data,"balsel_genes_expressed.txt"), header =T)
genes <- read.table(paste0(data,"balsel_genes_new.txt"), header =T)

disease_genes <- read.table(paste0(data,"REF_LA_filtered_annotation_disease.bed"))
colnames(disease_genes) <- c("chrom", "start", "end")

disease_genes$chrom <- factor(disease_genes$chrom)
levels(disease_genes$chrom)[1:5]<- c("A1", "A4", "A2", "XY", "A3")

disease_genes_balsel <- inner_join(disease_genes, genes, by = c("chrom", "start", "end"))
write.table(disease_genes_balsel, file=paste0(data,"disease_genes_balsel.txt"), row.names = FALSE, quote = FALSE)

```



## get max_CLR per gene for the whole genome

```{r}
window_size <- 7000
num_windows <- floor(nrow(scan) / window_size) # 706
results <- list()
for (i in 1:707) {
    start <- (i - 1) * window_size + 1
    end <- i * window_size
    window_df <- scan[start:end, ]
    
    result <- window_df %>% inner_join(gff, by = "chrom") %>%
        filter(physPos >= start & physPos <= end) 
    
    results[[i]] <- result}

result <- do.call(rbind, results) # 937846
length(results) # 318, half way thru
write.table(result, file=paste0(data,"balsel_sites_wholegenome.txt"), row.names = FALSE, quote = FALSE)
```




## gametophyte-biased genes have higher CLR than unbiased genes?

```{r}
group_by(bias, category) %>%
  dplyr::summarise(
    count = dplyr::n(),
    mean = mean(max_CLR, na.rm = TRUE),
    sd = sd(max_CLR, na.rm = TRUE)
  )

aov <- aov(max_CLR ~ category, data = bias)
summary(aov)

ggplot(bias, aes(x=category, y=max_CLR)) + 
  geom_boxplot() + ylim(0,250) +theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.ticks.x = element_blank(),panel.spacing = unit(0.1, "cm"),strip.background = element_blank(),strip.placement = "outside",legend.position ="none")+ theme(text = element_text(size = 15))
```
category count mean sd 
leaf_bias	242	54.55001	99.11744	
pollen_bias	142	52.56474	106.45978	
unbiased	21	150.48489	553.22779	

            Df   Sum Sq Mean Sq F value Pr(>F)  
category      2   186420   93210   3.715 0.0252 *
Residuals   402 10086918   25092                 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


## contigency test

category, p_biased, unbiased, l_biased
balancing_sel, 142, 21, 242
whole_genome, 6823, 822, 10301

```{r}
bias_tabler <- read.csv(paste0(data,"biased_genes.txt"))
bias_tabler
row.names(bias_tabler) <- bias_tabler[,1]
tablep <- bias_tabler[,2:3]
tablel <- bias_tabler[,3:4]
table <- bias_tabler[,c(2,4)]

#cnts <- read.table(paste0(directory,"biased_cnts.txt"), header =T)
#cnts <- cnts[1:4,]
cntsr <- gather(bias_tabler, group, count, p_biased:l_biased)
cntsr$group <- factor(cntsr$group)
levels(cntsr$group) <- c("sporophyte-biased", "gametophyte-biased", "unbiased")
cntsr$group <- factor(cntsr$group, level=c("gametophyte-biased", "sporophyte-biased", "unbiased"))


hcols <- c("black", "grey" ,"white")
p1 <- ggplot(cntsr) +
geom_col(aes(x = category, y = count, fill = group), position = "fill", colour ="black")  + xlab("") + ylab("Proportion of genes") + theme_classic() +ggtitle("a") + scale_fill_manual(values = hcols) + theme(axis.text=element_text(size=10), axis.title=element_text(size=10)) + guides(fill = guide_legend(override.aes = list(size = 10)))

p2 <- ggplot(cntsr) +
geom_col(aes(x = category, y = count, fill = group), colour ="black")  + xlab("") + ylab("Number of genes") + theme_classic() +ggtitle("b") + scale_fill_manual(values = hcols) + theme(axis.text=element_text(size=10), axis.title=element_text(size=10)) + guides(fill = guide_legend(override.aes = list(size = 10)))

p1 + p2 + plot_layout(guides = "collect") & theme(legend.position = "right")
```
real number + proportions


chi square test
```{r}
chisq.test(tablep)
chisq.test(tablel)
chisq.test(table)
```
	Pearson's Chi-squared test with Yates' continuity correction

data:  tablep
X-squared = 0.3143, df = 1, p-value = 0.5751


	Pearson's Chi-squared test with Yates' continuity correction

data:  tablel
X-squared = 0.0037144, df = 1, p-value = 0.9514


	Pearson's Chi-squared test with Yates' continuity correction

data:  table
X-squared = 1.1705, df = 1, p-value = 0.2793

fisher's exact test
```{r}
fisher.test(tablep)
fisher.test(tablel)
fisher.test(table)
```
	Fisher's Exact Test for Count Data

data:  tablep
p-value = 0.4533
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.5320146 1.4241511
sample estimates:
odds ratio 
 0.8503417 


	Fisher's Exact Test for Count Data
data:  tablel
p-value = 0.815
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.6298322 1.6396362
sample estimates:
odds ratio 
  1.041836 


	Fisher's Exact Test for Count Data

data:  table
p-value = 0.2685
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.7133327 1.0969436
sample estimates:
odds ratio 
 0.8858652 

