---
title: "FC - pi plot"
author: "Meng"
date: "2023-06-16"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(dplyr)
library(rattle)
library("gridExtra")
directory="/Users/yuanmeng/Library/CloudStorage/OneDrive-UniversityofToronto/Manuscripts/rumex_haploid_sel/pixy/"
```

## merge with pi data
```{r}
pixy <- read.table(paste0(directory,"pixy_pi_syn.0329.genewise.txt"), header=TRUE) 
pixy <- pixy %>% filter(no_sites>=50) 
colnames(pixy)[2:4] <- c("chrom", "start", "end") # 15048
pixy <- inner_join(pixy, gff, by = c("chrom","start","end"))

#DE_tissue<-as.data.frame(result_tissue)
DE_tissue <- read.table("/Users/yuanmeng/Library/CloudStorage/OneDrive-UniversityofToronto/Manuscripts/rumex_haploid_sel/deseq2/result_tissue_full.txt", header=T)
#DE_tissue <- left_join(DE_tissue,cnt,by="gene")
DE_tissue <- DE_tissue %>% filter(p_mean>5 | l_mean>5)

FC_pixy<- inner_join(DE_tissue,pixy,by="gene") # 11734
write.table(FC_pixy, file = paste0(directory,"FC_pixy_innerjoin.txt"), row.names = FALSE, quote = FALSE)
FC_pixy <- read.table(paste0(directory,"FC_pixy_innerjoin.txt"),header = TRUE)
#FC_pixy <- FC_pixy %>% filter(padj<0.1) #11203
```


## prepare gene sets
separate to p-biased and l-biased
then 4 quantiles within each category
DE gene cutoffs: padj<0.1 & baseMean >= 1
logFC>0: pollen biased, logFC <-: leaf biased
```{r}
pollen_bias <- FC_pixy %>% filter(log2FoldChange>0) # 4391, 4123
leaf_bias <- FC_pixy %>% filter(log2FoldChange<0) # 7344, 7080

bin<-binning(pollen_bias$log2FoldChange, bins=4, method="quantile",
                     labels=NULL, ordered=TRUE, weights=NULL)
bin<-binning(leaf_bias$log2FoldChange, bins=4, method="quantile",
                     labels=NULL, ordered=TRUE, weights=NULL)
plot(bin)
```

break and tags for FC
```{r}
breaks <- c(-30,-3.02,-2.16,-1.27,0,1.02,2.54,5.36,30)
# specify interval/bin labels
tags <- c("(,-3.02]","(-3.02,-2.16]", "(-2.16,-1.27]", "(-1.27,0)", "(0,1.02]", "(1.02,2.54]","(2.54,5.36]", "(5.36,)")

# bucketing values into bins
group_tags <- cut(FC_pixy$log2FoldChange, 
                  breaks=breaks, 
                  include.lowest=TRUE, 
                  right=FALSE, 
                  labels=tags)
# inspect bins
summary(group_tags)

ratio_groups <- factor(group_tags, 
                           levels = tags,
                           ordered = TRUE)
```
(,-3.02] (-3.02,-2.16] (-2.16,-1.27]     (-1.27,0)      (0,1.02]   (1.02,2.54] 
         1834          1837          1838          1835          1097          1105 
  (2.54,5.36]       (5.36,) 
         1093          1096 
pollen
1.02]
2.54]
5.36]
26.2]

leaf
-3.02]
-2.16]
-1.27]
-0.101]




## plot
```{r}
v <- FC_pixy %>% dplyr::select(log2FoldChange,avg_pi) 
vgroup <- as_tibble(FC_pixy) %>% 
  mutate(tag = case_when(
    log2FoldChange > 0 & log2FoldChange <= 1.02 ~ tags[5],
    log2FoldChange > 1.02 & log2FoldChange <= 2.54 ~ tags[6],
    log2FoldChange > 2.54 & log2FoldChange <= 5.36 ~ tags[7],
    log2FoldChange > 5.36   ~ tags[8],
    
    log2FoldChange <= -3.02 ~ tags[1],
    log2FoldChange > -3.02 & log2FoldChange <= -2.16 ~ tags[2],
    log2FoldChange > -2.16 & log2FoldChange <= -1.27 ~ tags[3],
    log2FoldChange > -1.27 & log2FoldChange < 0 ~ tags[4]

    ))
#summary(vgroup)

vgroup$tag <- factor(vgroup$tag,
                       levels = tags,
                       ordered = FALSE)

ggplot(data = vgroup, mapping = aes(x=tag,y=avg_pi)) + 
    geom_jitter(color="#1F78B4",alpha=0.2) +
    geom_boxplot(fill="#1F78B4",color="black",alpha=0.3) + 
    labs(x='log2FoldChange', y='Synonymous nucleotide diversity') +
    ggtitle('a')+
    guides(color=FALSE) + ylim(0,0.06)+
    theme_classic() + theme(text = element_text(size = 12))

```
 (,-3.02] (-3.02,-2.16] (-2.16,-1.27]     (-1.27,0)      (0,1.02]   (1.02,2.54] 
         1834          1837          1838          1835          1097          1105 
  (2.54,5.36]       (5.36,) 
         1093          1096 


## applying adjp<0.1 cutoff
```{r}
breaks <- c(-30,-3.05,-2.22,-1.39,0,1.25,2.76,5.65,30)
# specify interval/bin labels
tags <- c("(,-3.05]","(-3.05,-2.22]", "(-2.22,-1.39]", "(-1.39,0)", "(0,1.25]", "(1.25,2.76]","(2.76,5.65]", "(5.65,)")
# bucketing values into bins
group_tags <- cut(FC_pixy$log2FoldChange, 
                  breaks=breaks, 
                  include.lowest=TRUE, 
                  right=FALSE, 
                  labels=tags)
# inspect bins
summary(group_tags)
ratio_groups <- factor(group_tags, 
                           levels = tags,
                           ordered = TRUE)

v <- FC_pixy %>% select(log2FoldChange,avg_pi) 
vgroup <- as_tibble(v) %>% 
  mutate(tag = case_when(
    log2FoldChange > 0 & log2FoldChange <= 1.25 ~ tags[5],
    log2FoldChange > 1.25 & log2FoldChange <= 2.76 ~ tags[6],
    log2FoldChange > 2.76 & log2FoldChange <= 5.65 ~ tags[7],
    log2FoldChange > 5.65   ~ tags[8],
    log2FoldChange <= -3.05 ~ tags[1],
    log2FoldChange > -3.05 & log2FoldChange <= -2.22 ~ tags[2],
    log2FoldChange > -2.22 & log2FoldChange <= -1.39 ~ tags[3],
    log2FoldChange > -1.39 & log2FoldChange < 0 ~ tags[4]
    ))
#summary(vgroup)
vgroup$tag <- factor(vgroup$tag,
                       levels = tags,
                       ordered = FALSE)

ggplot(data = vgroup, mapping = aes(x=tag,y=avg_pi)) + 
    geom_jitter(color="#1F78B4",alpha=0.2) +
    geom_boxplot(fill="#1F78B4",color="black",alpha=0.3) + 
    labs(x='log2FoldChange', y='Synonymous nucleotide diversity') +
    ggtitle('a')+
    guides(color=FALSE) + ylim(0,0.06)+
    theme_classic() + theme(text = element_text(size = 12))
```
0,1.25]
2.76]
5.65]
26.2]
(-1.39
-1.39]
-2.22]
-3.05]
(,-3.05] (-3.05,-2.22] (-2.22,-1.39]     (-1.39,0)      (0,1.25]   (1.25,2.76] 
         1779          1754          1775          1772          1024          1037 
  (2.76,5.65]       (5.65,) 
         1031          1031 



## pi - log2FC regression 
```{r}
p1 <- ggplot(FC_pixy, aes(x=log2FoldChange, y=avg_pi)) + geom_point(colour="#1F78B4", alpha=0.2) + geom_smooth(color="black", method = lm) + theme_classic() + theme(text = element_text(size = 10))+ labs(x='log2FoldChange', y='nucleotide diversity (syn)') + ggtitle('a')

p3 <- ggplot(FC_pixy, aes(x=abs(log2FoldChange), y=avg_pi)) + geom_point(colour="#1F78B4", alpha=0.2) + geom_smooth(color="black", method = lm) + theme_classic() + theme(text = element_text(size = 10))+ labs(x='abs(log2FoldChange)', y='nucleotide diversity (syn)') + ggtitle('c')

grid.arrange(p1, p2, p3, p4, nrow = 2)
```

linear regression
```{r}
cor(FC_pixy$avg_pi, FC_pixy$log2FoldChange) # 0.04935226
cor(FC_pixy$avg_pi, abs(FC_pixy$log2FoldChange)) # 0.06233079

model <- lm(avg_pi ~ log2FoldChange, data = FC_pixy)
summary(model)

model <- lm(avg_pi ~ abs(log2FoldChange), data = FC_pixy)
summary(model)
```

lm(formula = avg_pi ~ log2FoldChange, data = FC_pixy)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.009455 -0.005023 -0.001796  0.002677  0.098571 

Coefficients:
                Estimate Std. Error t value Pr(>|t|)    
(Intercept)    8.034e-03  7.034e-05 114.218  < 2e-16 ***
log2FoldChange 9.707e-05  1.814e-05   5.352 8.85e-08 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.007619 on 11733 degrees of freedom
Multiple R-squared:  0.002436,	Adjusted R-squared:  0.002351 
F-statistic: 28.65 on 1 and 11733 DF,  p-value: 8.847e-08

Call:
lm(formula = avg_pi ~ abs(log2FoldChange), data = FC_pixy)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.010146 -0.005001 -0.001787  0.002662  0.098493 

Coefficients:
                     Estimate Std. Error t value Pr(>|t|)    
(Intercept)         7.535e-03  1.022e-04  73.703  < 2e-16 ***
abs(log2FoldChange) 1.783e-04  2.636e-05   6.765  1.4e-11 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.007614 on 11733 degrees of freedom
Multiple R-squared:  0.003885,	Adjusted R-squared:  0.0038 
F-statistic: 45.76 on 1 and 11733 DF,  p-value: 1.399e-11

##
```{r}
calculate_pi <- function(data) {
  data <- data %>%
    mutate(pi_syn = avg_pi * no_sites)
  
  sum_pi_syn <- sum(data$pi_syn)
  sum_no_sites <- sum(data$no_sites)
  weighted_pi_syn <- sum_pi_syn / sum_no_sites

  return(weighted_pi_syn)
}

pi_values <- vgroup %>%
  group_by(tag) %>%
  summarise(weighted_pi_syn = calculate_pi(cur_data()))
write.table(pi_values, file=paste0(directory,"pi_values_weighted.txt"), sep = "\t",row.names = FALSE, quote = FALSE)

pi_values <- read.table(paste0(directory,"pi_values_weighted.txt"), header=T)

ggplot(data = pi_values, mapping = aes(x=log2FoldChange, y=weighted_pi_s)) + 
    geom_point()+
    theme_classic() + theme(text = element_text(size = 12))

ggplot(data = pi_values, mapping = aes(x=log2FoldChange, y=pi_s_mean)) + 
    geom_point()+ 
    theme_classic() + theme(text = element_text(size = 12))

table(vgroup$tag)
```
 bootstrap 1000 times with in bins


## pi_n / pi_s and expression bias
removed zero in pi_s pi_n
```{r}
tags <- c("(,-3.02]","(-3.02,-2.16]", "(-2.16,-1.27]", "(-1.27,0)", "(0,1.02]", "(1.02,2.54]","(2.54,5.36]", "(5.36,)")
statsr_group <- as_tibble(statsr) %>% 
  mutate(tag = case_when(
    log2FoldChange > 0 & log2FoldChange <= 1.02 ~ tags[5],
    log2FoldChange > 1.02 & log2FoldChange <= 2.54 ~ tags[6],
    log2FoldChange > 2.54 & log2FoldChange <= 5.36 ~ tags[7],
    log2FoldChange > 5.36   ~ tags[8],
    
    log2FoldChange <= -3.02 ~ tags[1],
    log2FoldChange > -3.02 & log2FoldChange <= -2.16 ~ tags[2],
    log2FoldChange > -2.16 & log2FoldChange <= -1.27 ~ tags[3],
    log2FoldChange > -1.27 & log2FoldChange < 0 ~ tags[4]
    ))

calculate_pi <- function(data) {
  data <- data %>%
    mutate(pi_syn = avg_pi * no_sites.x) %>%
    mutate(pi_nonsyn = avg_pi_nonsyn * no_sites.y)
  sum_pi_syn <- sum(data$pi_syn)
  sum_pi_nonsyn <- sum(data$pi_nonsyn)
  sum_no_sites_x <- sum(data$no_sites.x)
  sum_no_sites_y <- sum(data$no_sites.y)
  weighted_pi_syn <- sum_pi_syn / sum_no_sites_x
  weighted_pi_nonsyn <- sum_pi_nonsyn / sum_no_sites_y
  mean_avg_pi <- mean(data$avg_pi)
  mean_avg_pi_nonsyn <- mean(data$avg_pi_nonsyn)
  result <- data.frame(
    sum_pi_syn = sum_pi_syn,
    sum_pi_nonsyn = sum_pi_nonsyn,
    sum_no_sites_x = sum_no_sites_x,
    sum_no_sites_y = sum_no_sites_y,
    mean_avg_pi = mean_avg_pi,
    mean_avg_pi_nonsyn = mean_avg_pi_nonsyn,
    weighted_pi_syn = weighted_pi_syn,
    weighted_pi_nonsyn = weighted_pi_nonsyn,
    ratio_weighted_pis = weighted_pi_nonsyn / weighted_pi_syn
  )
  return(result)
}

write.table(statsr_group, file=paste0(directory,"statsr_group.txt"), row.names = FALSE, quote = FALSE)
write.table(pi_values, file=paste0(directory,"pi_values.txt"), row.names = FALSE, quote = FALSE)
write.table(pi_values_old, file=paste0(directory,"pi_values_old.txt"), sep = "\t",row.names = FALSE, quote = FALSE)

pi_values_old <- vgroup %>%
  group_by(tag) %>%
  summarise(mean = mean(avg_pi), sd = sd(avg_pi))

# for each tag, run calculate_pi()
pi_values <- statsr_group %>%
  group_by(tag) %>%
  summarise(stats = list(calculate_pi(cur_data())))%>%
  unnest(stats)

table(statsr_group$tag)

ggplot(data = pi_values, mapping = aes(x=tag,y=ratio_weighted_pis)) + 
    geom_point()+
    theme_classic() + theme(text = element_text(size = 12))
```



